frontend/src/features/settings/account/ExportAllDataForm.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import Typography from '@material-ui/core/Typography';
+import { useSnackbar } from 'notistack';
+
+import { showErrorAlert } from '../../../utils/notifications';
+
+import type { ApiRequestOptions } from 'src/services/openapi/core/ApiRequestOptions';
+import { OpenAPI } from 'src/services/openapi';
+
+// The method below replaces the auto-generated usersMeExportsAllRetrieve
+// from services/openapi/core/request.ts because it does
+// not support returning response.blob which we need for
+// downloading files from the API endpoint /users/me/exports/all
+async function usersMeExportsAllRetrieveBlob(): Promise<Blob> {
+  const api_url = process.env.REACT_APP_API_URL;
+  const options: ApiRequestOptions = {
+    method: 'GET',
+    path: '/users/me/exports/all/',
+  };
+  const access_token =
+    typeof OpenAPI.TOKEN === 'function'
+      ? await OpenAPI.TOKEN(options)
+      : OpenAPI.TOKEN;
+  const headers = { Authorization: `Bearer ${access_token}` };
+  const response = await fetch(`${api_url}${options.path}`, {
+    credentials: 'include',
+    method: options.method,
+    headers: headers,
+  });
+  return response.blob();
+}
+
+const ExportAllDataForm = () => {
+  const [loading, setLoading] = useState(false);
+  const [downloadUrl, setDownloadUrl] = useState('');
+  const { enqueueSnackbar } = useSnackbar();
+
+  const downloadExport = async () => {
+    const zipFileBlob = await usersMeExportsAllRetrieveBlob();
+    setDownloadUrl(window.URL.createObjectURL(zipFileBlob));
+    setLoading(false);
+  };
+
+  const prepareExport = () => {
+    setLoading(true);
+    setDownloadUrl('');
+    try {
+      downloadExport();
```

Comment:
As `downloadExport()` is async, we need to await the promise to be able to catch potential errors.  
Otherwise any error in the async call would cause an unhandled rejection, and would never be caught by the "catch" block.

```suggestion
  const prepareExport = async () => {
    setLoading(true);
    setDownloadUrl('');
    try {
      await downloadExport();
```